libs/frontend/feature-build/src/lib/dnd/drag-drop.ts

```diff
@@ -41,16 +52,10 @@ const checkSelfDrop = (
   return false
 }
 
-export const onDrag = (
-  e: DragEvent,
-  context: ISiteContext,
-  dragSrc: IDraggedComponent,
-  targetIndex: number,
-  target: IComponent,
-  elementRef: HTMLElement | undefined,
-  // Only check for top&bottom hover
-  verticalOnly?: boolean,
-): IDropProps => {
+export const onDrag = (options: IOnDragOptions): IDropProps => {
```

Comment:
For [#864](https://github.com/samatechtw/pubstudio/issues/864), I noticed that the drop actually works under some circumstances (in some scenarios it really didn't work!), it's just that we have to move the cursor slightly to the side.

https://github.com/samatechtw/pubstudio-builder/assets/59676941/ee960b22-607b-4f6b-99ee-2177dc9e0aec

This happens because we check for hoverSelf first, then top/bottom and left/right. The easiest solution I could think of is to rearrange the order to make it check for top/bottom and left/right first, then hoverSelf.

I also made some small changes to the check condition so that dropping a component to the same level as its parent can work properly. It seems to work now, feel free to let me know if you notice any problem!

Reply:
Makes sense to me 👍